text_collector_examples/apcaccess.sh

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env bash
+#
+# Description: Expose metrics from apcaccess
+#
+# Author: Erwin Oegema <erwin.oegema@gmail.com>
+# Required tools:
+# - apcaccess
+# - bc
+# - awk
+# - cut
+
+# Make sure we also terminate on pipe operations
+set -uo pipefail
+
+# It is recommended to check https://linux.die.net/man/8/apcaccess for
+# more properties that could be exposed
+# Example response:
+#
+# APC      : 001,000,0000
+# DATE     : 2018-10-03 15:36:29 +0200
+# HOSTNAME : hostname
+# VERSION  : 3.14.14 (31 May 2016) debian
+# UPSNAME  : backups700
+# CABLE    : USB Cable
+# DRIVER   : USB UPS Driver
+# UPSMODE  : Stand Alone
+# STARTTIME: 2018-09-27 18:28:59 +0200
+# MODEL    : Back-UPS ES 700G
+# STATUS   : ONLINE
+# LINEV    : 234.0 Volts
+# LOADPCT  : 10.0 Percent
+# BCHARGE  : 100.0 Percent
+# TIMELEFT : 27.6 Minutes
+# MBATTCHG : 5 Percent
+# MINTIMEL : 3 Minutes
+# MAXTIME  : 0 Seconds
+# SENSE    : Medium
+# LOTRANS  : 180.0 Volts
+# HITRANS  : 266.0 Volts
+# ALARMDEL : 30 Seconds
+# BATTV    : 13.6 Volts
+# LASTXFER : Unacceptable line voltage changes
+# NUMXFERS : 1
+# XONBATT  : 2018-10-01 14:20:44 +0200
+# TONBATT  : 0 Seconds
+# CUMONBATT: 10 Seconds
+# XOFFBATT : 2018-10-01 14:20:54 +0200
+# STATFLAG : 0x05000008
+# SERIALNO : xxx
+# BATTDATE : 2018-04-21
+# NOMINV   : 230 Volts
+# NOMBATTV : 12.0 Volts
+# FIRMWARE : 871.O4 .I USB FW:O4
+# END APC  : 2018-10-03 15:37:04 +0200
+
+function write_line {
+	local key="$1"
+	local val="$2"
+	local type="$3"
+	local help="$4"
+
+	echo '# HELP' "$key" "$help"
+	echo '# TYPE' "$key" "$type"
+	echo "$key" "$val" "$data_date"
```

Comment:
The timestamp must be removed, it will cause errors when using this with the node_exporter.

Reply:
Well, I've been running this script for a day and I do get the data without issues

Reply:
You are possibly running an older version of the node_exporter. Timestamps in the textfile collector are considered an error now.

Reply:
For more context, exposing timestamps from metrics is generally a bad practice, since it causes unexpected staleness and gaps. We prefer to let Prometheus assign the timestamp of the metric, since that's when the data was seen by Prometheus.

To provide a "when was this updated" method for textfiles, we have a metric in the node_exporter called `node_textfile_mtime_seconds`.